mainhandler/handlerequests.go

```diff
@@ -85,25 +85,31 @@ func (mainHandler *MainHandler) HandleWatchers(ctx context.Context) {
 		}
 	}()
 
-	watchHandler := watcher.NewWatchHandler()
+	watchHandler, err := watcher.NewWatchHandler(ctx)
 
-	if err := watchHandler.Initialize(ctx); err != nil {
+	if err != nil {
 		logger.L().Ctx(ctx).Error(err.Error(), helpers.Error(err))
 		return
 	}
 
-	newImageIDs, err := watchHandler.GetImageIDsForSBOMCalculation()
+	newImageIDs, err := watchHandler.ListImageIDsNotInStorage()
 	if err != nil {
 		logger.L().Ctx(ctx).Error(err.Error(), helpers.Error(err))
 	}
 
 	// get commands for scanning new images
 	commandsList := []*apis.Command{}
 	for _, imageID := range newImageIDs {
-		for _, wlid := range watchHandler.GetWlidsForImageID(imageID) {
-			cmd := buildScanCommandForWorkload(ctx, wlid, watchHandler.GetContainerToImageIDForWlid(wlid), apis.TypeCalculateSBOM)
-			commandsList = append(commandsList, cmd)
+		wlidsForImage := watchHandler.GetWlidsForImageID(imageID)
+		if len(wlidsForImage) == 0 {
+			logger.L().Ctx(ctx).Error(fmt.Sprintf("Image %s is not used by any workload", imageID))
```

Comment:
is this really an error?

Reply:
yes, I don't see how we can have an image in the map and not have its workloads